src/commands/remember/remember-count.js

```diff
@@ -1,5 +1,42 @@
 const {ApplicationCommandOptionType} = require('discord.js');
-const {saveNumberMessages} = require("../../utils/rememberMessages");
+const {parseMessage, addMessages} = require("../../utils/rememberMessages");
+const {defaultExcludeBotMessages} = require('../../../config.json');
+const {getNumberMessages} = require('../../utils/apiCalls');
+
+const getMessagesAndReturnId = async(messagesToSave, channel, num, excludeBotMessages, startId) =>{
+    let messageObjArray;
+    if(startId){
+         //get the first 100 messages at a specifc point
+         messageObjArray = await getNumberMessages(channel, num, startId)
+    }
+    else{
+        //get the num messages starting with the latest one
+        messageObjArray = await getNumberMessages(channel, num)
+    }
+
+    if (!messageObjArray) {
+        return;
+    };
+
+    //sort the message in ascending order of timestamp
+    messageObjArray.sort((a, b) => new Date(a.timestamp) - new Date(b.timestamp));
+
+    //go through the entire array and save messages, maybe excluding bot
+    //m is a kvp [1] is the actual data
+    for (const m of messageObjArray) {
+        startId = m[1].id;
+
+        // exlude bot messages if option is enabled
+        if (excludeBotMessages && m[1].author.bot) {
+            continue;
+        }
+
+        const message = parseMessage(m[1])
+        messagesToSave.push(message)
+    }
```

Comment:
This should be an equivalent alternative
```js
messagesToSave = messageObjArray
    .filter(([_, msg]) => !(excludeBotMessages && msg.author.bot))
    .map(([_, msg]) => parseMessage(msg));
startId = messageObjArray.at(-1)[1].id; // last element
```

Reply:
To be clear, I strongly prefer data-oriented approaches like this (where everything basically becomes array processing, transforming and filtering and modifying arrays into arrays into arrays). My suggestion is definitely denser, which may come at a cost of readability, so if I suggest an "alternative" I'm hoping you'll just do whatever you think works best (take or leave my approach).

Reply:
The main reason I didn't implement this was the messageObjArray wasn't iterable normally, both filter and map threw errors. And I couldn't get startId the way you suggested as id was marked as undefined. I was in a rush cause of my car appointment so I can take a closer look later.

Reply:
Could try `[...messageObjArray]`